lib/explorer/backend/lazy_series.ex

```diff
@@ -350,6 +351,26 @@ defmodule Explorer.Backend.LazySeries do
     Backend.Series.new(data, dtype)
   end
 
+  @impl true
+  def select(%Series{} = pred, %Series{} = on_true, %Series{} = on_false) do
+    args = [
+      series_or_lazy_series!(pred),
+      series_or_lazy_series!(on_true),
+      series_or_lazy_series!(on_false)
+    ]
+
+    data = new(:select, args, aggregations?(args), window_functions?(args))
+
+    dtype =
+      if on_true.dtype != on_false.dtype do
```

Comment:
I think they can be different if they are numeric. So if the `on_true` is integer and `on_false` is float, the final dtype must be `float`.

Reply:
good point, i'll leverage some of the helper functions to get this right